snowflake/pkg/snowflake/snowflake.go

```diff
@@ -101,3 +105,36 @@ func (c *client) DropWarehouse(ctx context.Context, name string) error {
 	_, err := c.db.ExecContext(ctx, query)
 	return err
 }
+
+func (c *client) UseDatabase(ctx context.Context, name string) error {
+	query := fmt.Sprintf("USE DATABASE %s", name)
+	c.logger.V(2).Info("Snowflake query", "query", query)
+	_, err := c.db.ExecContext(ctx, query)
+	return err
+}
+
+func (c *client) UseSchema(ctx context.Context, name string) error {
+	query := fmt.Sprintf("USE SCHEMA %s", name)
+	c.logger.Info("Snowflake query", "query", query)
+	_, err := c.db.ExecContext(ctx, query)
+	return err
+}
+
+func (c *client) StageFile(ctx context.Context, path string, stage string) error {
+	query := fmt.Sprintf("PUT file://%s @%s AUTO_COMPRESS = FALSE OVERWRITE = TRUE", path, stage)
+	c.logger.Info("Snowflake query", "query", query)
+	_, err := c.db.ExecContext(ctx, query)
+	return err
+}
+
+func (c *client) ExecMultiStatementQuery(ctx context.Context, query string, result bool) (*sql.Rows, error) {
+	multi_statement_context, _ := gosnowflake.WithMultiStatement(ctx, 0)
+	c.logger.Info("Snowflake query", "query", query)
+	if !result {
+		_, err := c.db.ExecContext(multi_statement_context, query)
+		return nil, err
+	} else {
+		rows, err := c.db.QueryContext(multi_statement_context, query)
+		return rows, err
+	}
```

Comment:
do you think we could have 2 different functions for the 2 different cases?

Reply:
Sounds good to me, changed.